lib/structs/query/query_output_details.py

```diff
@@ -12,5 +12,18 @@ class QueryOutputDetails:
     and how to output them to_html(), to_string() etc.
     """
 
-    properties_to_select: Optional[Set[PropEnum]] = None
+    properties_to_select: Optional[List[PropEnum]] = None
     output_type: Optional[QueryOutputTypes] = None
+
+    @staticmethod
+    def from_kwargs(prop_cls: PropEnum, **kwargs):
+        """
+        method to parse and create this dataclass from kwargs
+        :param prop_cls a PropEnum class to get enum from for properties to select
+        :param kwargs: A set of kwargs to parse and use to set attributes to dataclass
+        """
+        props = {prop_cls.from_string(prop) for prop in kwargs["properties_to_select"]}
```

Comment:
Are the kwargs a nested dict? I can't think of a case where they can pass the same name twice in properties_to_select (or maybe the set should be moved upwards)

The list comprehension could then just be done here, rather than going to a set, then a list

Reply:
sure